commands/daemon/settings.go

```diff
@@ -144,3 +145,35 @@ func (s *SettingsService) Write(ctx context.Context, req *rpc.WriteRequest) (*rp
 	}
 	return &rpc.WriteResponse{}, nil
 }
+
+// Delete removes a key from the config file
+func (s *SettingsService) Delete(ctx context.Context, req *rpc.DeleteRequest) (*rpc.DeleteResponse, error) {
+	toDelete := req.GetKey()
+
+	// Check if settings key actually existing, we don't use Viper.InConfig()
+	// since that doesn't check for keys formatted like daemon.port or those set
+	// with Viper.Set(). This way we check for all existing settings for sure.
+	keyExists := false
+	keys := []string{}
+	for _, k := range configuration.Settings.AllKeys() {
+		if !strings.HasPrefix(k, toDelete) {
+			keys = append(keys, k)
+			continue
+		}
+		keyExists = true
+	}
+
+	if !keyExists {
+		return nil, errors.New(tr("key not found in settings"))
+	}
+
+	updatedSettings := viper.New()
+	for _, k := range keys {
+		updatedSettings.Set(k, configuration.Settings.Get(k))
+	}
+
+	if err := updatedSettings.WriteConfigAs(req.GetFilePath()); err != nil {
+		return nil, err
+	}
```

Comment:
We have `SettingsService.Write` to do this operation, usually the other operations perform an update of the running environment (that is lost when the daemon terminates if not saved).

Could it be possible to replace the `configuration.Settings` with the `updatedSettings` at runtime?

Reply:
It is not possible to delete a key from `configuration.Settings` at runtime, but we could set them as `""` as a workaround. Maybe something like this:
```
for _, k := range configuration.Settings.AllKeys() {
		if !strings.HasPrefix(k, toDelete) {
			keys = append(keys, k)
			continue
		}
		keyExists = true
		configuration.Settings.Set(k, "")
	}
```
I'm not using `SettingsService.Write` because it relies on `configuration.Settings` and the key would not be deleted as intended.

Reply:
Overwriting it is 👍